src/main/java/com/nvidia/spark/rapids/jni/GpuTimeZoneDB.java

```diff
@@ -55,18 +60,24 @@ public class GpuTimeZoneDB {
   private static Map<String, Integer> zoneIdToTable;
 
   // use this reference to indicate if time zone cache is initialized.
-  private static HostColumnVector fixedTransitions;
+  private static HostColumnVector transitions;
+  // initial year set to 1900 because some transition rules start early
+  private static final int initialTransitionYear = 1900;
+  private static long maxTimestamp;
+  private static final ZoneId utcZoneId = ZoneId.of("UTC");
 
   /**
    * This should be called on startup of an executor.
    * Runs in a thread asynchronously.
    * If `shutdown` was called ever, then will not load the cache
    */
-  public static void cacheDatabaseAsync() {
+  public static void cacheDatabaseAsync(int maxYear) {
     // start a new thread to load
     Runnable runnable = () -> {
       try {
-        cacheDatabaseImpl();
+        cacheDatabaseImpl(maxYear);
+        maxTimestamp = LocalDateTime.of(maxYear+1, 1, 2, 0, 0, 0)
```

Comment:
```suggestion
        maxTimestamp = LocalDateTime.of(maxYear, 12, 30, 0, 0, 0)
```
Considering the timezone offset is max 18 hours, use a less value 12-30 instead of 12-31 to make sure GPU has the transitions for the max timestamp.

Reply:
I'd prefer we support the whole maxYear, so it's easier for the user to understand that any timestamp that has the year maxYear or less is supported. An analysis of all timezones an timezone rules show that there is transition rule happening in the first week of January, except for some fixed ones happening before 2100.